My Project/clsTranslateWinForms.vb

```diff
@@ -1 +1,638 @@
+    ''' <summary>   
+    '''     Recursively traverses the <paramref name="clsMenuItems"/> menu hierarchy and returns a 
+    '''     string containing the parent, name and associated text of each (sub)menu option in 
+    '''     <paramref name="clsMenuItems"/>. The string is formatted as a comma-separated list 
+    '''     suitable for importing into a database.
+    ''' </summary>
+    '''
+    ''' <param name="clsControl">        The WinForm control that is the parent of the menu. </param>
+    ''' <param name="clsMenuItems">     The WinForm menu items to add to the return string. </param>
+    '''
+    ''' <returns>   
+    '''     A string containing the parent and name of each (sub)menu option in
+    '''     <paramref name="clsMenuItems"/>. The string is formatted as a comma-separated list
+    '''     suitable for importing into a database. </returns>
+    '''--------------------------------------------------------------------------------------------
+    Public Shared Function GetMenuItemsAsCsv(clsControl As Control, clsMenuItems As ToolStripItemCollection) As String
+        If IsNothing(clsControl) OrElse IsNothing(clsMenuItems) Then
+            Return ""
+        End If
+
+        Dim dctComponents As Dictionary(Of String, Component) = New Dictionary(Of String, Component)
+        GetDctComponentsFromMenuItems(clsMenuItems, dctComponents)
+
+        Dim strMenuItemsAsCsv As String = ""
+        For Each clsComponent In dctComponents
+
+            If TypeOf clsComponent.Value Is ToolStripItem Then
+                Dim clsMenuItem As ToolStripItem = DirectCast(clsComponent.Value, ToolStripItem)
+                strMenuItemsAsCsv &= clsControl.Name & "," & clsComponent.Key & "," & GetCsvText(clsMenuItem.Text) & vbCrLf
+            Else
+                MsgBox("Developer Error: Translation dictionary entry (" & clsControl.Name & "," & clsComponent.Key & ") contained unexpected value type.")
+            End If
+
+        Next
+        Return strMenuItemsAsCsv
+    End Function
+
+    '''--------------------------------------------------------------------------------------------
+    ''' <summary>
+    '''    Populates dictionary <paramref name="dctComponents"/> with the control 
+    '''    <paramref name="clsControl"/> and its children.    
+    '''    The dictionary can then be used to conveniently translate the control text (see other
+    '''    functions and subs in this class).
+    ''' </summary>
+    '''
+    ''' <param name="clsControl">       The control used to populate the dictionary. </param>
+    ''' <param name="dctComponents">    [in,out] Dictionary to store the control and its children. 
+    '''                                 </param>
+    '''--------------------------------------------------------------------------------------------
+    Private Shared Sub GetDctComponentsFromControl(clsControl As Control,
+                                                   ByRef dctComponents As Dictionary(Of String, Component),
```

Comment:
@lloyddewit this subroutine fills the dictionary passed into it, I would lean towards a name like `FillDctComponentsFromControl` rather than the `GetDctComponentsFromControl`, to me the later entails returning a new dictionary from the passed control.

Reply:
agreed, changed